### PR TITLE
E2E: Add ICA MsgSubmitTx tests (success + failure)

### PR DESCRIPTION
**Pull request by chatton**
## Description

This PR adds an E2E test which tests the happy path for in interchain accounts transaction using the `icad` docker images.

Thank you @charleenfei and @damiannolan for all the help with this one 🥇 

closes: #1941

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: e2e/interchain_accounts_test.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+// RegisterICA will attempt to register an interchain account on the counterparty chain.
+func (s *InterchainAccountsTestSuite) RegisterICA(ctx context.Context, chain *cosmos.CosmosChain, user *ibctest.User, fromAddress, connectionID string) error {
+	version := "" // allow app to handle the version as appropriate.
+	msg := intertxtypes.NewMsgRegisterAccount(fromAddress, connectionID, version)
+	txResp, err := s.BroadcastMessages(ctx, chain, user, msg)
+	s.AssertValidTxResponse(txResp)
+	return err
+}
+
+func (s *InterchainAccountsTestSuite) TestMsgSubmitTx_SuccessfulTransfer() {
```
[the remark this answers is not in the document]
Comment (chatton) [reply e2e/interchain_accounts_test.go:44]: separated out into 2 tests as you suggested!

File: e2e/interchain_accounts_test.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+		})
+
+		t.Run("verify tokens transferred", func(t *testing.T) {
+			balance, err := chainB.GetBalance(ctx, chainBAccount.Bech32Address(chainB.Config().Bech32Prefix), chainB.Config().Denom)
+			s.Require().NoError(err)
+
+			_, err = chainB.GetBalance(ctx, hostAccount, chainB.Config().Denom)
+			s.Require().NoError(err)
+
+			expected := testvalues.IBCTransferAmount + testvalues.StartingTokenAmount
```
Comment (colin-axner) [inline e2e/interchain_accounts_test.go:132]: I find it a little confusing that we use `IBCTransferAmount` in a bank send. Maybe we could add a short comment explaining we use that var in the bank send. I don't have a strong preference though

File: e2e/interchain_accounts_test.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+				chainA,
+				controllerAccount,
+				submitMsg,
+			)
+
+			s.AssertValidTxResponse(resp)
+			s.Require().NoError(err)
+		})
+
+		t.Run("verify balance is the same", func(t *testing.T) {
```
Comment (colin-axner) [inline e2e/interchain_accounts_test.go:212]: The test above adds a wait for 10 blocks. Maybe we should add an assertion that the packet was relayed?

**Review by colin-axner — APPROVED** [review]
LGTM! Excellent work

File: e2e/interchain_accounts_test.go (hunk 3 above)
Comment (chatton) [reply e2e/interchain_accounts_test.go:212]: Great suggestion, will add this.

File: e2e/interchain_accounts_test.go [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+	)
+	suite.Run(t, new(InterchainAccountsTestSuite))
+}
+
+type InterchainAccountsTestSuite struct {
+	testsuite.E2ETestSuite
+}
+
+// RegisterICA will attempt to register an interchain account on the counterparty chain.
+func (s *InterchainAccountsTestSuite) RegisterICA(ctx context.Context, chain *cosmos.CosmosChain, user *ibctest.User, fromAddress, connectionID string) error {
```
Comment (damiannolan) [inline e2e/interchain_accounts_test.go:36]: I think I'd prefer `RegisterInterchainAccount` to be explicit.

We could refactor this function to take the `MsgRegisterAccount` as an arg in place of the `fromAddr` and `connnectionID`. Then we can built the message with an explicit version for both incentivized and non-incentivized channels.

File: e2e/interchain_accounts_test.go [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+				ToAddress:   chainBAccount.Bech32Address(chainB.Config().Bech32Prefix),
+				Amount:      sdk.NewCoins(testvalues.DefaultTransferAmount(chainB.Config().Denom)),
+			}
+
+			// assemble submitMessage tx for intertx
+			submitMsg, err := intertxtypes.NewMsgSubmitTx(
+				transferMsg,
+				connectionId,
+				controllerAccount.Bech32Address(chainA.Config().Bech32Prefix),
+			)
```
Comment (damiannolan) [inline e2e/interchain_accounts_test.go:106]: naming nits: I think we should name vars to correspond with the msg type. I got confused with `transferMsg` as when I read it I think ibc transfer (ics20).

What do you think about using `msgSend` and `msgSubmitTx`, I find it reads a little easier.

File: e2e/interchain_accounts_test.go [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+	// setup relayers and connection-0 between two chains
+	// channel-0 is a transfer channel but it will not be used in this test case
+	relayer, _ := s.SetupChainsRelayerAndChannel(ctx)
+	chainA, chainB := s.GetChains()
+
+	connectionId := "connection-0"
+
+	// setup 2 accounts: controller account on chain A, a second chain B account.
+	// host account will be created when the ICA is registered
+	controllerAccount := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
```
Comment (seantking) [inline e2e/interchain_accounts_test.go:57]: wdyt about renaming `ControllerAccount` to `OwnerAccount` and the `hostAccount` as `InterchainAccount`?

cc: @damiannolan

File: e2e/interchain_accounts_test.go [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+		hostAccount, err = s.QueryInterchainAccount(ctx, chainA, controllerAccount.Bech32Address(chainA.Config().Bech32Prefix), connectionId)
+		s.Require().NoError(err)
+		s.Require().NotZero(len(hostAccount))
+
+		channels, err := relayer.GetChannels(ctx, s.GetRelayerExecReporter(), chainA.Config().ChainID)
+		s.Require().NoError(err)
+		s.Require().Equal(len(channels), 2)
+	})
+
+	t.Run("execute bank transfer over ICA through controller account", func(t *testing.T) {
```
Comment (seantking) [inline e2e/interchain_accounts_test.go:81]:
```suggestion
	t.Run("interchain account executes a bank transfer on behalf of the corresponding owner account", func(t *testing.T) {
```

File: e2e/interchain_accounts_test.go [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+
+		t.Run("fund host wallet", func(t *testing.T) {
```
Comment (seantking) [inline e2e/interchain_accounts_test.go:83]:
```suggestion
		t.Run("fund interchain account wallet", func(t *testing.T) {
```

File: e2e/testconfig/testconfig.go [hunk 9]
```diff
@@ -8,6 +8,7 @@ import (
 )
 
 const (
+
```
Comment (seantking) [inline e2e/testconfig/testconfig.go:11]:
```suggestion
```

**Review by seantking — APPROVED** [review]
Left a few nits. Nice work :-)

File: e2e/interchain_accounts_test.go (hunk 4 above)
Comment (chatton) [reply e2e/interchain_accounts_test.go:36]: good idea makes sense

File: e2e/interchain_accounts_test.go (hunk 6 above)
Comment (charleenfei) [reply e2e/interchain_accounts_test.go:54]: i personally think that this change would be more confusing -- but that's just my opinion

Comment (colin-axner) [reply e2e/interchain_accounts_test.go:54]: I personally like host/controller terminology

**Pull request merged**